### PR TITLE
fix(coverage): Error if the emit cache is invalid

### PR DESCRIPTION
**Pull request by sigmaSd**
fix https://github.com/denoland/deno/issues/16846

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Can you provide a test?

File: cli/tests/coverage_tests.rs [hunk 1]
```diff
@@ -26,6 +26,62 @@ mod coverage {
...
+      .join("mod_before.ts");
+    let mod_after_path = util::testdata_path()
+      .join(&invalid_cache_path)
+      .join("mod_after.ts");
+    let mod_path = util::testdata_path()
+      .join(&invalid_cache_path)
+      .join("mod.ts");
+
+    // Write the inital mod.ts file
+    std::fs::copy(mod_before_path, &mod_path).unwrap();
```
Comment (dsherret) [inline cli/tests/coverage_tests.rs:48]: Could we copy this to the temp directory instead? That will also allow us to remove mod.ts from git.

File: cli/tools/coverage/mod.rs [hunk 2]
```diff
@@ -677,7 +678,8 @@ pub async fn cover_files(
       | MediaType::Mts
       | MediaType::Cts
       | MediaType::Tsx => {
-        match ps.emit_cache.get_emit_code(&file.specifier, None) {
+        let source_hash = get_source_hash(&file.source, ps.emit_options_hash);
+        match ps.emit_cache.get_emit_code(&file.specifier, Some(source_hash)) {
```
Comment (dsherret) [inline cli/tools/coverage/mod.rs:682]: This makes sense. While we're here, we could change this second parameter from `Option<u64>` to just `u64` to force people to provide a hash everywhere.

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks for fixing this @sigmaSd.

@dsherret are you happy this this solution?

**Review by dsherret — APPROVED** [review]
Yup. Thanks @sigmaSd!

**Pull request merged**